libdispatch/dattput.c

```diff
@@ -83,9 +83,9 @@ be assigned or ::NC_GLOBAL for a global attribute.
 \param name Attribute \ref object_name. \ref attribute_conventions may
 apply.
 
-\param len Number of values provided for the attribute.
+\param len Length of `value` not including trailing null.
```

Comment:
This is not correct. The trailing null may be included for C users.

Reply:
That is how i typically used it -- I would pass `strlen(text)+1` as the length, but then I grepped the NetCDF source and saw that typically `strlen(text)` was passed as the argument.   Not sure which is correct...

Reply:
Neither is correct. It's up to the user.

Fortran does not use a NULL, so you might say don't bother writing one.

But then C programs must add a NULL every time they read the attribute. So if writing a bunch of netCDF files that will be read by a bunch of C programs, you might say always include the NULL.

So it is left up to the user. This may be unwise, but that's the way it has been and cannot be changed without breaking lots of code for someone. ;-)